.github/copilot-instructions.md

```diff
@@ -0,0 +1,249 @@
+    - All tests must pass (zero failures).
+    - Any non-zero exit code from test commands is considered a failure.
+
+- **Workflow:**
+    - On success: Report completion
+    - Otherwise: Report error(s) with logs for diagnostics.
+        - Collect logs from `artifacts/log/` and the console output for both build and test steps.
+        - Attach relevant log files or error snippets when reporting failures.
+
+---
+
+## 3. CoreCLR (CLR) Workflow
+
+From the repository root:
+
+- Build:
+  `./build.sh -subset clr`
+
+- Run tests:
+  `cd src/tests && ./build.sh && ./run.sh`
+
+- More info:
+    - [Building CoreCLR Guide](https://github.com/dotnet/runtime/blob/main/docs/workflow/building/coreclr/README.md)
+    - [Building and Running CoreCLR Tests](https://github.com/dotnet/runtime/blob/main/docs/workflow/testing/coreclr/testing.md)
+
+---
+
+## 4. Mono Runtime Workflow
+
+From the repository root:
+
+- Build:
+  `./build.sh -subset mono+libs`
+
+- Run tests:
+
+  ```bash
+  ./build.sh clr.host
+  cd src/tests
+  ./build.sh mono debug /p:LibrariesConfiguration=debug
+  ./run.sh
+  ```
+
+- More info:
+    - [Building Mono](https://github.com/dotnet/runtime/blob/main/docs/workflow/building/mono/README.md)
+    - [Running test suites using Mono](https://github.com/dotnet/runtime/blob/main/docs/workflow/testing/mono/testing.md)
+
+---
+
+## 5. Libraries Workflow
+
+From the repository root:
+
+- Build:
+  `./build.sh -subset libs -rc release`
+
+- Run tests:
+  `./build.sh -subset libs.tests -test -rc release`
+
+- More info:
+    - [Build Libraries](https://github.com/dotnet/runtime/blob/main/docs/workflow/building/libraries/README.md)
+    - [Testing Libraries](https://github.com/dotnet/runtime/blob/main/docs/workflow/testing/libraries/testing.md)
+
+### 5.1. How To: Identify Affected Libraries
+
+For each changed file under `src/libraries/`, find the matching library and its test project(s).
+Most libraries use:
+
+- Source: `src/libraries/<LibraryName>/src/<LibraryName>.csproj`
+
+- Tests (single):
+    - `src/libraries/<LibraryName>/tests/<LibraryName>.Tests.csproj`
+    - OR `src/libraries/<LibraryName>/tests/<LibraryName>.Tests/<LibraryName>.Tests.csproj`
+
+- Tests (multiple types):
+    - `src/libraries/<LibraryName>/tests/FunctionalTests/<LibraryName>.Functional.Tests.csproj`
+    - `src/libraries/<LibraryName>/tests/UnitTests/<LibraryName>.Unit.Tests.csproj`
+    - Or similar.
+
+---
+
+### 5.2. How To: Build and Test Specific Library
+
+If only one library is affected:
+
+1. **Navigate to the library directory:**
+   `cd src/libraries/<LibraryName>`
+
+2. **Build the library:**
+   `dotnet build`
+
+3. **Build and run all test projects:**
+
+    - For each discovered `*.Tests.csproj` in the `tests` subdirectory:
+      `dotnet build /t:test ./tests/<TestProject>.csproj`
+
+        - *Adjust path as needed. If in doubt, search with `find tests -name '*.csproj'`.*
+
+    - `dotnet build /t:test` is generally preferred over `dotnet test`
```

Comment:
This was fixed last year, but I checked and it does appear to have crept back in  https://github.com/dotnet/runtime/issues/94183